libsql-server/src/auth/parsers.rs

```diff
@@ -160,4 +173,117 @@ mod tests {
         let out = parse_http_basic_auth_arg("always").unwrap();
         assert_eq!(out, Some("always".to_string()));
     }
+
+    const EXAMPLE_JWT_PUBLIC_KEY: &str = include_str!("../../assets/test/auth/example1.pem");
+    const EXAMPLE_JWT_PRIVATE_KEY: &str = include_str!("../../assets/test/auth/example1.key");
+    const EXAMPLE_JWT: &str = include_str!("../../assets/test/auth/example1.jwt");
+    const EXAMPLE3_JWT: &str = include_str!("../../assets/test/auth/example3.jwt");
+    const MULTI_JWT_PUBLIC_KEY: &str = include_str!("../../assets/test/auth/combined123.pem");
+
+    #[test]
+    fn parse_jwt_key_single_pem() {
+        let key = parse_jwt_key(EXAMPLE_JWT_PUBLIC_KEY);
+        assert!(matches!(key, Ok(DecodingKeyContainer::Single(_))));
+        match key {
+            Ok(DecodingKeyContainer::Single(key)) => {
+                let mut validation = jsonwebtoken::Validation::new(jsonwebtoken::Algorithm::EdDSA);
+                validation.required_spec_claims.remove("exp");
+                let decoded = jsonwebtoken::decode::<Token>(&EXAMPLE_JWT, &key, &validation);
+
+                assert!(matches!(
+                    decoded,
+                    Ok(jsonwebtoken::TokenData {
+                        header: _,
+                        claims: _
+                    })
+                ));
+
+                let jsonwebtoken::TokenData { header, claims } = decoded.unwrap();
+                assert_eq!(header.alg, jsonwebtoken::Algorithm::EdDSA);
+
+                let claims = format!("{:?}", claims);
+                assert_eq!(claims, "Token { id: None, a: None, p: Some(Authorized { read_only: Some(Scopes { namespaces: Some({example1c, example1a, example1b}), tags: None }), read_write: None, read_only_attach: None, read_write_attach: None, ddl_override: None }), exp: Some(+33713-01-29T21:47:33Z) }");
```

Comment:
std::fmt::Debug does not guarantee the ordering of fields
maybe better to make assertions on individual claims fields
that way we can ignore fields that we're not concerned about
and such individual assertions could be easily reusable

Reply:
I opened up (crate pub) the `p` field from `Token` in order to validate that. Not sure if that's going to be a point of concern or not.